### ISSUE TITLE
Cannot create property '$isUnderneathDocArray' on string 'nested'

### ISSUE THREAD

**Issue by chaeron** [Automattic/mongoose#10226]
Getting this with the latest 5.12.7 point release:

```
/home/andrzej/Dropbox/Shore/TGLN/dev/migration/node_modules/mongoose/lib/schema.js:782
      schemaType.schema.singleNestedPaths[key].$isUnderneathDocArray = true;
      
TypeError: Cannot create property '$isUnderneathDocArray' on string 'nested'
    at Schema.path (/home/andrzej/Dropbox/Shore/TGLN/dev/migration/node_modules/mongoose/lib/schema.js:782:70)
    at Schema.add (/home/andrzej/Dropbox/Shore/TGLN/dev/migration/node_modules/mongoose/lib/schema.js:503:12)
    at new Schema (/home/andrzej/Dropbox/Shore/TGLN/dev/migration/node_modules/mongoose/lib/schema.js:129:10)

```

5.12.5 is fine.

Aren't point releases supposed to not have breaking changes?

WTF?

**Comment by n-e** [comment]
Hi,

With mongoose 5.12.8 and the repro below I get the same error as OP :

```js
const mongoose = require('mongoose');

new mongoose.Schema({
  a: [
    {
      b: {
        type: new mongoose.Schema({
          c: {
            d: Number,
          },
        }),
      },
    },
  ],
});
```

There is no error if I make c a number instead of an object.

**Comment by vkarpov15** [comment]
Looks like a duplicate of #10193, should be fixed in 5.12.9 :+1:
